### DISCUSSION TITLE
Citi Hackathon GitProxy Use Case: Deploy push protections and policies on top of Git

### DISCUSSION THREAD

**Discussion by lucaborella89** [finos/finos-pmcs#328] [Ideas]
# Deploy push protections and policies on top of Git
![logo](https://github.com/user-attachments/assets/e4ada461-a321-4412-b249-2869207aadad)

GitProxy enables companies such as Citi, Royal Bank of Canada (RBC) and NatWest to scan code for security and compliance issues before it is pushed to a codebase. Today, GitProxy is limited in the checks and types of scans it runs before allowing code to be pushed upstream or allowed outside of a private network. We want hackers to create discrete functions that takes a git diff as input and outputs detected security holes and compliance issues in the code. Ask yourself, what are the types of checks & scans that organisations like Citi should run on code before publishing it on the Internet.

## Expected Features
This may includes:
1. Detection of data (incl. test data, JSONs, .log, .csv, .xlsx)
2. Usage of non-standard cryptography and encryption techniques
3. Patentable assets detection (incl. images, videos, graphics, designs)
4. Detection of AI/ML usage (incl. OSS libraries, weights etc.)

## Useful apps, systems, projects, technologies, libraries, datasets: 
GitProxy is an open source library hosted on the public internet on GitHub
https://github.com/finos/git-proxy

**Comment by JamieSlome** [comment]
For more detail on the use case, please refer to: 

[Hack-a-thon: Go Plugin Crazy 🎉](https://github.com/finos/git-proxy/discussions/754)
